spec/dry/types/default_spec.rb

```diff
@@ -165,4 +165,80 @@
       end
     end
   end
+
+  describe '#to_s' do
+    context 'static valute' do
+      subject(:type) { Dry::Types['string'].default('foo') }
+
+      it 'returns string representation of the type' do
+        expect(type.to_s).to eql('#<Dry::Types[Default<Definition<String> value="foo">]>')
+      end
+    end
+
+    context 'callable value' do
+      subject(:type) { Dry::Types['string'].default(value_constructor) }
+
+      context 'proc' do
+        let(:value_constructor) { proc { 'foo' } }
+
+        let(:line_no) { value_constructor.source_location[1] }
+
+        it 'returns string representation of the type' do
+          expect(type.to_s).
+            to eql(
+              "#<Dry::Types[Default<Definition<String> "\
+              "value_fn=spec/dry/types/default_spec.rb:#{ line_no }>]>"
+            )
+        end
+      end
+
+      context 'proc w/o source' do
+        let(:value_constructor) { method(:Integer).to_proc }
+
+        it 'returns string representation of the type' do
+          expect(type.to_s).
+            to eql(
+              "#<Dry::Types[Default<Definition<String> "\
+              "value_fn=(lambda)>]>"
+            )
+        end
+      end
+
+      context 'method' do
+        let(:value_constructor) { Kernel.method(:Integer) }
+
+        it 'returns string representation of the type' do
+          expect(type.to_s).
+            to eql(
+              "#<Dry::Types[Default<Definition<String> "\
+              "value_fn=Kernel.Integer>]>"
+            )
+        end
+      end
+
+      context 'callable object' do
+        let(:value_constructor) do
+          obj = Object.new
+
+          def obj.to_s
+            "callable"
+          end
+
+          def obj.call(*)
+            5
+          end
+
+          obj
+        end
+
+        it 'returns string representation of the type' do
+          expect(type.to_s).
+            to eql(
+              "#<Dry::Types[Default<Definition<String> "\
+              "value_fn=callable.call>]>"
```

Comment:
Would it make more sense for this to be `callable#call`?

Reply:
technically, it's `.call`. Hashtag is for class methods, i.e. `SomeCallableClass#call`, it'd be wrong to do `"#{ callable.class }#call"` since we rely on the instance's interface

Reply:
> Hashtag is for class methods

@flash-gordon are you _sure?_ For the longest time I've used `#` as the delimiter when denoting a class' _instance_ methods. As far as I can tell, this is in accordance with community convention, e.g. on https://docs.ruby-lang.org/en/2.6.0/String.html you'll see instance methods prefixed by `#`:

<img width="471" alt="screen shot 2019-03-08 at 9 44 49 pm" src="https://user-images.githubusercontent.com/3134/54024276-6cccec80-41eb-11e9-85de-1adea69e308b.png">

Reply:
oh wait, I think I understand what you're saying here: because we're not actually referring to the _class_, we can't use the class-based notation for its instance methods (e.g. `Class#method`)... is that right?

Reply:
@timriley sorry, I didn't make it clear. Here we rely on the fact that the object producing default values responds to `:call`. It doesn't mean it has the `call` method defined in its class, that was my point. However, now I think in this extreme case we can call `.method(:call)` (bad pun) and see where it's actually present. Then print different output depending on where its definition is.

Reply:
@flash-gordon Cool, I fully understand now, thanks for the discussion :) And your idea to take this one step further sounds good, too. 👍